ppo.ipynb

```diff
@@ -0,0 +1,352 @@
+    "id": "FHET6yCkw4ds"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Visualization functions"
+   ],
+   "metadata": {
+    "id": "E7_OrJQ0qls7"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "def display_video(frames, filename=\"temp.mp4\", frame_repeat=1):\n",
+    "    \"\"\"Save and display video.\"\"\"\n",
+    "    # Write video\n",
+    "    with imageio.get_writer(filename, fps=60) as video:\n",
+    "        for frame in frames:\n",
+    "            for _ in range(frame_repeat):\n",
+    "                video.append_data(frame)\n",
+    "    # Read video and display the video\n",
+    "    video = open(filename, \"rb\").read()\n",
+    "    b64_video = base64.b64encode(video)\n",
+    "    video_tag = (\n",
+    "        '<video  width=\"320\" height=\"240\" controls alt=\"test\" '\n",
+    "        'src=\"data:video/mp4;base64,{0}\">'\n",
+    "    ).format(b64_video.decode())\n",
+    "    return IPython.display.HTML(video_tag)"
+   ],
+   "outputs": [],
+   "metadata": {
+    "id": "SOhF8c-AqlTa"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Definitions\n",
+    "\n",
+    "Definition of all the parts used in the learning loop: environment, agent, etc."
+   ],
+   "metadata": {
+    "id": "Q-40V1lKDlip"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "# Configuration\n",
+    "ENV_NAME = \"Pendulum-v1\""
+   ],
+   "outputs": [],
+   "metadata": {
+    "id": "y7uMVaYaFcA0"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "# Create an environment, grab the spec\n",
+    "# We should use our own environment here\n",
+    "environment = helpers.make_environment(ENV_NAME)\n",
+    "environment_spec = specs.make_environment_spec(environment)\n",
+    "\n",
+    "# environment = PendulumEnv()\n",
+    "# environment_spec = specs.make_environment_spec(environment)"
+   ],
+   "outputs": [],
+   "metadata": {
+    "id": "rltCiwH2FYWA"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "# Create the networks\n",
+    "# We should use our own netowkrs here\n",
+    "# agent_networks = dm_ppo.make_gym_networks(environment_spec)"
+   ],
+   "outputs": [],
+   "metadata": {
+    "id": "kfJrCivxHmqH"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "# Create the agent\n",
+    "# We should use PPO agent\n",
+    "agent = RandomAgent(environment_spec)"
+   ],
+   "outputs": [],
+   "metadata": {
+    "id": "xpJvcAOtF2t3"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Interaction loop"
+   ],
+   "metadata": {
+    "id": "kluuB3xPJd_u"
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "def training_loop(\n",
+    "    environment,\n",
+    "    agent,\n",
+    "    num_episodes=None,\n",
+    "    num_steps=None,\n",
```

Comment:
To keep it simple, I think we can just use `num_episodes` and forget about `num_steps`. What do you think @emasquil ?

Reply:
Do you mean that we'll always run all episodes until termination?